src/tests/ftest/control/dmg_telemetry_io_basic.py

```diff
@@ -0,0 +1,132 @@
+                                    if value < metrics_data[key-1]\
+                                        [name][host][rank][target][size]:
+                                        status = False
+                                        invalid += " *value decreased"
+                                self.log.info(
+                                    "    %-9s %-12s %-4s %-6s %-6s %s %s",
+                                    testloop, host, rank, target, size, value,
+                                    invalid)
+        if not status:
+            self.fail("##Telemetry test io metrics verification failed.")
+
+    def display_io_test_metrics(self, metrics_data):
+        """ Display metrics_data.
+
+        Args:
+            metrics_data (dict): a dictionary of host keys linked to a
+                                 list of io metric names.
+        """
+        for key in sorted(metrics_data):
+            self.log.info(
+                "\n  %12s: %s",
+                "Initial " if key == 0 else "Test Loop {}".format(key),
+                metrics_data[key])
+
+
+    def test_telmetry_metrics(self):
+        """JIRA ID: DAOS-5241
+
+            Create files of 500M and 1M with transfer size 1M to verify the
+            DAOS engine IO telemetry basic metrics infrastructure.
+
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=control,telemetry
+        :avocado: tags=test_with_telemetry_basic,test_io_telemetry
+        :avocado: tags=test_io_telemetry_basic
+
+        """
+        block_sizes = self.params.get("block_sizes", "/run/*")
+        transfer_sizes = self.params.get("transfer_sizes", "/run/*")
+        test_metrics = self.params.get("io_test_metrics", "/run/*")
+        threshold = self.params.get("io_test_metrics_valid", "/run/*")
+        i = 0
+        self.add_pool(connect=False)
+        self.add_container(pool=self.pool)
+        metrics_data = {}
+        for block_size in block_sizes:
+            for transfer_size in transfer_sizes:
+                metrics_data[i] = self.telemetry.get_io_metrics()
+                i += 1
+                self.log.info("==Start ior testloop: %s, Block Size = %s, "
+                              "transfer_size =  %s", i, block_size,
+                              transfer_size)
+                self.ior_cmd.block_size.update(block_size)
+                self.ior_cmd.transfer_size.update(transfer_size)
+                test_file_suffix = "_{}".format(i)
+                # Run ior command.
+                try:
+                    self.run_ior_with_pool(
+                        timeout=200, create_pool=False, create_cont=False,
+                        test_file_suffix=test_file_suffix)
+                except TestFail:
+                    self.log.info("#ior command failed!")
+        metrics_data[i] = self.telemetry.get_io_metrics()
+        self.display_io_test_metrics(metrics_data)
```

Comment:
I wonder if it would be better if you integrate the display in the verify function, such that the display is also sorted the way you are doing the verification.

Reply:
From this PR earlier, I just query, display and verify the selected 13 telemetry IO metrics. 
I updated the script to display the entire IO metrics per Phil's comment. Since this is the very first IO telemetry metrics script, it is ok to query and display the entire telemetry IO metrics and to verify the selected telemetry IO metrics.  The main objective to validate the telemetry infrastructure.

Reply:
I think this can be avoided if we split ENGINE_IO_METRICS into logical areas, which will also help solve the problem of redefining the metrics in yaml file

Reply:
I think because this is one of the earlier telemetry test, it is even more important to do this right, as it will likely be used as a reference for subsequent telemetry test

Reply:
Please see new commit for address all of the comments.